src/server/api/lib/message-sending.js

```diff
@@ -1,4 +1,222 @@
+ *
+ * @param {number} organizationId The ID of organization
+ */
+export const getMessagingServiceCandidates = async organizationId => {
+  const { rows: messagingServiceCandidates } = await r.knex.raw(
+    `
+      select
+        messaging_service.messaging_service_sid,
+        count(messaging_service_stick.messaging_service_sid) as count
+      from messaging_service
+      left join messaging_service_stick
+        on messaging_service_stick.messaging_service_sid = messaging_service.messaging_service_sid
+      where
+        messaging_service.organization_id = ?
+      group by
+        messaging_service.messaging_service_sid
+      order by count desc
+    `,
+    [organizationId]
+  );
+  return messagingServiceCandidates;
+};
+
+/**
+ * Assign an appropriate messaging service for a (cell, organization) pairing.
+ * This creates a messaging_service_stick record.
+ *
+ * TODO: Update logic to allow for per-campaign decisions.
+ *
+ * @param {string} cell An E164-formatted destination cell phone number
+ * @param {number} organizationId The ID of the organization to create the mapping for
+ * @returns {string} The (S)ID of the messaging service assigned to that (cell, organization)
+ */
+export const assignMessagingServiceSID = async (cell, organizationId) => {
+  const {
+    rows: [{ messaging_service_sid }]
+  } = await r.knex.raw(
+    `
+      with chosen_messaging_service_sid as (
+        select
+          messaging_service.messaging_service_sid,
+          count(messaging_service_stick.messaging_service_sid) as count
+        from messaging_service
+        left join messaging_service_stick
+          on messaging_service_stick.messaging_service_sid = messaging_service.messaging_service_sid
+          and messaging_service.organization_id = ?
+        group by
+          messaging_service.messaging_service_sid
+        order by count asc
+        limit 1
+      )
+      insert into messaging_service_stick (cell, organization_id, messaging_service_sid)
+      values (?, ?, (select messaging_service_sid from chosen_messaging_service_sid))
+      returning messaging_service_sid;
+    `,
+    [organizationId, cell, organizationId]
+  );
+
+  return messaging_service_sid;
+};
+
+/**
+ * Fetch messaging service by its ID
+ * @param {string} messagingServiceId The messaging service ID
+ * @returns {object} The messaging service record if found, or undefined;
+ */
+export const getMessagingServiceById = async messagingServiceId =>
+  r
+    .knex("messaging_service")
+    .where({ messaging_service_sid: messagingServiceId })
+    .first();
+
+/**
+ * Fetches an existing assigned messaging service for a campaign contact. If no messaging service
+ * has been assigned then assign one and return that.
+ * @param {number} campaignContactId The ID of the target campaign contact
+ * @returns {object} Assigned messaging service Postgres row
+ */
+export const getContactMessagingService = async campaignContactId => {
```

Comment:
```suggestion
export const getContactMessagingService = async (campaignContactId, organizationId) => {
```

This could get passed the organizationId from [here](https://github.com/politics-rewired/Spoke/pull/311/files#diff-5d117a0afdd02753cbd75a8890706ee4R168), saving 1 round trip to DB

Reply:
It makes sense to fetch here because you need to switch on it for what service you use to send, right? But organization ID will always have been already queried in this request lifecycle